Assets/Scripts/Effects/Description.cs

```diff
@@ -0,0 +1,232 @@
+        DiscardDecievers = false;
+    }
+
+    // Figure out how much damage / healing / armor an effect is going to affect
+    private static void GetStats(Effect effect, Stats lowBound, Stats highBound, int currentStreak)
+    {
+        (float low, float high) = effect.EffectRange(currentStreak);
+        switch (effect.type)
+        {
+            case EffectType.Barrier:
+                lowBound.Armor += low;
+                highBound.Armor += high;
+                break;
+            case EffectType.Damage:
+            case EffectType.DamageIgnoreBarrier:
+            case EffectType.DamageOnDeceiverInDiscardPile:
+            case EffectType.DiscardAllWithPerCardDamage:
+                lowBound.Damage += low;
+                highBound.Damage += high;
+                break;
+            case EffectType.Heal:
+                lowBound.Damage -= high;
+                highBound.Damage -= low;
+                break;
+            default:
+                // Unimplemented
+                break;
+        }
+        
+    }
+
+    string DescribeRange(float low, float high)
+    {
+        if (Mathf.Approximately(low, high))
+        {
+            return $"{ low}";
+        }
+        else
+        {
+            return $"{low} - {high}";
+        }
+    }
+
+    // Generate reasonable text 
+    public override string ToString()
+    {
+        string EnemyDamageStr = $"{DescribeRange(EnemyStatsLowBound.Damage, EnemyStatsHighBound.Damage)} dmg to an enemy";
+        string SelfDamageStr = $"{DescribeRange(SelfStatsLowBound.Damage, SelfStatsHighBound.Damage)}dmg to you";
+        string SelfHealStr = $"{DescribeRange(-SelfStatsLowBound.Damage, -SelfStatsHighBound.Damage)} health";
+        string EnemyHealStr = $"{DescribeRange(-EnemyStatsLowBound.Damage, - EnemyStatsHighBound.Damage)} health";
+        string SelfArmorStr = (SelfStatsLowBound.Armor > 0 ? "Gain " : "Lose") + 
+            $"{DescribeRange(SelfStatsLowBound.Armor, SelfStatsHighBound.Armor)} armor";
+        string EnemyArmorStr = "Enemy " + 
+            (EnemyStatsLowBound.Armor > 0 ? "gains " : "loses" +
+            $"{DescribeRange(EnemyStatsLowBound.Armor, EnemyStatsHighBound.Armor)} armor");
+        string DiscardDecieverStr = "Discard all Deciever cards.";
+        string desc = "";
+
+        if (SelfStatsLowBound.Damage > 0)
+        {
+            desc += "Deal " + SelfDamageStr;
+            if (EnemyStatsLowBound.Damage > 0)
+            {
+                desc += " and " + EnemyDamageStr;
+            }
+            desc += ".";
+        }
+        else
+        {
+            if (EnemyStatsLowBound.Damage > 0)
+            {
+                desc += "Deal " + EnemyDamageStr + ".";
+            }
+        }
+        if (SelfStatsHighBound.Damage < 0)
+        {
+            desc += "Gain " + SelfHealStr;
+            if (EnemyStatsHighBound.Damage < 0)
+            {
+                desc += "and enemy gains" + EnemyHealStr + ".";
```

Comment:
Some of the strings are missing spaces that making the description super weird

Reply:
Fixed now